service/telemetry/config.go

```diff
@@ -5,14 +5,27 @@ package telemetry // import "go.opentelemetry.io/collector/service/telemetry"
 
 import (
 	"fmt"
+	"net"
+	"strconv"
 	"time"
 
 	"go.opentelemetry.io/contrib/config"
 	"go.uber.org/zap/zapcore"
 
 	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/featuregate"
 )
 
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+var disableAddressFieldForInternalTelemetryFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	"telemetry.disableAddressFieldForInternalTelemetry",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterFromVersion("v0.110.0"),
+	featuregate.WithRegisterToVersion("v0.110.0"),
```

Comment:
probably need a version much further into the future

Reply:
Done.